packages/sdk-codegen/src/python.gen.ts

```diff
@@ -263,9 +278,28 @@ ${this.hooks.join('\n')}
   }
 
   initArg(indent: string, property: IProperty) {
-    return `${indent}self.${property.name} = ${property.name}`
+    let propName = property.name
+    if (this.pythonKeywords.has(propName)) {
+      propName = propName + '_'
+    }
+    return `${indent}self.${propName} = ${propName}`
+  }
+
+  typeProperties(type: IType) {
+    return Object.values(type.requiredProperties).concat(Object.values(type.optionalProperties))
   }
 
+  private isBareForwardRef = (prop: IProperty) => {
+    // contains a "bare" forward reference e.g. `result_format: "ResultFormat"`
+    // so we need to emit the `__annotations__` property
+    return (
+      prop.required &&
+      (prop.type.customType || prop.type.className == 'EnumType') &&
```

Comment:
I believe `(prop.type.customType || prop.type.className == 'EnumType')` captures all the generated models/enums - i.e. none of the IntrinsicType

Reply:
I used 
```ts
prop.type instanceof EnumType 
```
as the check for the other enum generators. FWIW, I do preserve the base type of the `EnumType` as the element type so we know whether the enum is a string or an int. The Looker spec **only** has string enums, though.

Reply:
so should I do this?

```suggestion
      (prop.type.customType || prop.type instanceof EnumType) &&
```

Reply:
That would work. I don't think you even need the customType test because it won't be an EnumType unless it is an enum

Reply:
The goal is to catch model types like "User" as well as enum types like "ResultFormat" but not int/str/bool etc types. Can I do `!(prop.type instanceof IntrinsicType)` ?

Reply:
if you're trying to ignore just intrinsic types, yes. You can use `instanceof` or `!type.intrinsic`